back/engines/commercial/bulk_import_ideas/app/services/bulk_import_ideas/import_ideas_service.rb

```diff
@@ -0,0 +1,216 @@
+    end
+
+    def add_body_multiloc(idea_row, idea_attributes)
+      raise Error.new 'bulk_import_ideas_blank_body', row: idea_row[:id] if idea_row[:body_multiloc].blank?
+
+      idea_attributes[:body_multiloc] = idea_row[:body_multiloc]
+    end
+
+    def add_project(idea_row, idea_attributes)
+      raise Error.new 'bulk_import_ideas_blank_project', row: idea_row[:id] if idea_row[:project_title].blank?
+
+      project_title = idea_row[:project_title].downcase.strip
+      # Later iteration: Only load necessary attributes? Preload all projects before importing all?
+      project = Project.all.find do |find_project|
+        find_project
+          .title_multiloc
+          .values
+          .map { |v| v.downcase.strip }
+          .include? project_title
+      end
+      if project.blank?
+        raise Error.new 'bulk_import_ideas_project_not_found', value: idea_row[:project_title], row: idea_row[:id]
+      end
+
+      idea_attributes[:project] = project
+    end
+
+    def add_author(idea_row, idea_attributes)
+      raise Error.new 'bulk_import_ideas_blank_email', row: idea_row[:id] if idea_row[:user_email].blank?
+
+      author = User.find_by_cimail idea_row[:user_email]
+      raise Error.new 'bulk_import_ideas_email_not_found', value: idea_row[:user_email], row: idea_row[:id] if author.blank?
+
+      idea_attributes[:author] = author
+    end
+
+    def add_published_at(idea_row, idea_attributes)
+      return if idea_row[:published_at].blank?
+
+      published_at = nil
+      begin
+        published_at = Date.parse idea_row[:published_at]
+      rescue StandardError => _e
+        raise Error.new 'bulk_import_ideas_publication_date_invalid_format', value: idea_row[:published_at], row: idea_row[:id]
+      end
+
+      idea_attributes[:published_at] = published_at
+    end
+
+    def add_publication_status(_idea_row, idea_attributes)
+      idea_attributes[:publication_status] = 'published'
+    end
+
+    def add_location(idea_row, idea_attributes)
+      if (lat = idea_row[:latitude]&.to_f) && (lon = idea_row[:longitude]&.to_f)
+        location_point = {
+          'type' => 'Point',
+          'coordinates' => [lon, lat]
+        }
+        idea_attributes[:location_point_geojson] = location_point
+      end
+      idea_attributes[:location_description] = idea_row[:location_description] if idea_row[:location_description]
+    end
+
+    def add_phase(idea_row, idea_attributes)
+      return if idea_row[:phase_rank].blank?
+
+      phase_rank = idea_row[:phase_rank].to_i
+      project_phases = Phase.where(project: idea_attributes[:project])
+      if phase_rank > project_phases.size
+        raise Error.new 'bulk_import_ideas_maximum_phase_rank_exceeded', value: phase_rank, row: idea_row[:id]
+      end
+
+      phase = project_phases.order(:start_at).all[phase_rank - 1]
+      raise Error.new 'bulk_import_ideas_project_phase_not_found', value: phase_rank, row: idea_row[:id] if phase.blank?
```

Comment:
I think it can be simplified, because accessing a non-existing array index returns `nil`:
```suggestion
      raise Error.new 'bulk_import_ideas_project_phase_not_found', value: phase_rank, row: idea_row[:id] if phase
```